kagi/static/kagi/webauthn.js

```diff
@@ -1,329 +1,281 @@
-function b64enc(buf) {
-    return base64js.fromByteArray(buf)
-                   .replace(/\+/g, "-")
-                   .replace(/\//g, "_")
-                   .replace(/=/g, "");
-}
-
-function b64RawEnc(buf) {
-    return base64js.fromByteArray(buf)
-    .replace(/\+/g, "-")
-    .replace(/\//g, "_");
-}
-
-function hexEncode(buf) {
-    return Array.from(buf)
-                .map(function(x) {
-                    return ("0" + x.toString(16)).substr(-2);
-				})
-                .join("");
-}
-
-async function fetch_json(url, options) {
-    const response = await fetch(url, options);
-    const body = await response.json();
-    if (body.fail)
-        throw body.fail;
-    return body;
-}
-
-if (typeof window.Kagi === 'undefined') {
-  let Kagi = window.Kagi || {
-    begin_activate: '/kagi/api/begin-activate/',
-    begin_assertion: '/kagi/api/begin-assertion/',
-    verify_credential_info: '/kagi/api/verify-credential-info/',
-    verify_assertion: '/kagi/api/verify-assertion/',
-    keys_list: '/kagi/keys/',
-  };
-  console.error("window.Kagi is not defined, falling back to default URLs", Kagi);
+/* Licensed under the Apache License, Version 2.0 (the "License");
```

Comment:
Where is the Apache license coming all of a sudden? If this code is copied from somewhere else it might make sense to reference it. Otherwise it would be good to stick to one license in the project (which seems to be the simplified 2-clause BSD). @Natim can you shed some light on this?

Reply:
If memory serves, this came via the excellent work done to implement MFA for the Warehouse project: https://github.com/pypi/warehouse/blob/main/warehouse/static/js/warehouse/utils/webauthn.js

I agree that it makes sense to reference it. 👍

Reply:
Added via 9f7281a.

Reply:
Indeed — thanks, Raphaël. I added similar annotations in the missing spots via 4101e13.